connect/api/src/main/java/org/apache/kafka/connect/health/TaskState.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.health;
+
+import java.util.Objects;
+
+/**
+ * {@link TaskState} provides the state, ids and any errors associated with Connector
+ * tasks.
+ */
+public class TaskState extends AbstractState implements Comparable<TaskState> {
+
+    private final int taskId;
+
+    /**
+     * Provides an instance of {@link TaskState}
+     * @param taskId - the id associated with the connector task
+     * @param state  - the status of the task. Can't be NULL or EMPTY.
+     * @param workerId - id of the worker the task is associated with. Can't be NULL or EMPTY.
```

Comment:
For consistency, use `; may not be null or empty` in parameter descriptions. Also, no `-` after the parameter name since JavaDoc already handles this. Fix these everywhere.